ETBezierHandle.m

```diff
@@ -77,21 +81,25 @@ - (id) initWithActionHandler: (ETActionHandler *)anHandler
 		{
 			// FIXME: ugly
 			case NSCurveToBezierPathElement:
-				[handles addObject: AUTORELEASE([[ETBezierHandle alloc] initWithActionHandler: [ETBezierPointActionHandler sharedInstance]
-				                                                            manipulatedObject: aTarget
-																			         partcode: [path partcodeForControlPoint: 0 ofElement: i]])];
-				[handles addObject: AUTORELEASE([[ETBezierHandle alloc] initWithActionHandler: [ETBezierPointActionHandler sharedInstance]
-				                                                            manipulatedObject: aTarget
-																			         partcode: [path partcodeForControlPoint: 1 ofElement: i]])];
-				[handles addObject: AUTORELEASE([[ETBezierHandle alloc] initWithActionHandler: [ETBezierPointActionHandler sharedInstance]
-				                                                            manipulatedObject: aTarget
-																			         partcode: [path partcodeForControlPoint: 2 ofElement: i]])];
+                [handles addObject: AUTORELEASE([[ETBezierHandle alloc] initWithActionHandler: [ETBezierPointActionHandler sharedInstanceForObjectGraphContext: aContext]
```

Comment:
I also wrapped the first case in a loop to eliminate the unnecessary duplication, possibly resolving the FIXME.